File: implement-shell-tools/wc/wcl*.js
const fs = require("fs");
const path = require("path");

const folder = "sample-files";

const dir = path.join(__dirname, folder);
let total = 0;

fs.readdirSync(dir).forEach((f) => {
  const p = path.join(dir, f);
  if (!fs.statSync(p).isFile()) return;

  const lines = fs.readFileSync(p, "utf-8").split("\n").length;
  total += lines;

  console.log(lines, f);
});

console.log(total, "total");

File: implement-shell-tools/wc/wc-l*.py
from pathlib import Path

folder = "sample-files"
dir_path = Path(__file__).parent / folder

total = 0

for file in dir_path.iterdir():
    if not file.is_file():
        continue

    content = file.read_text(encoding="utf-8")
    lines = len(content.split("\n"))
    total += lines

    print(lines, file.name)

print(total, "total")